parallel_spectral_fitting/parallel_spectral_fitting.ipynb

```diff
@@ -0,0 +1,980 @@
+    "        # to fit to (so >number of parameters in the model for unregularized fits)\n",
+    "\n",
+    "        # The 1D Gaussian model with initial guesses for parameters\n",
+    "        g_init = models.Gaussian1D(amplitude=guesses[0][y_full, x_full],\n",
+    "                                   mean=guesses[1][y_full, x_full],\n",
+    "                                   stddev=guesses[2][y_full, x_full])\n",
+    "\n",
+    "        # If the initial guesses have any NaNs, also skip the fitting.\n",
+    "        if np.isnan(g_init.parameters).any():\n",
+    "            continue\n",
+    "\n",
+    "        # And fit with the Levenberg-Marquardt algorithm and least squares statistic.\n",
+    "        fit_g = fitting.LevMarLSQFitter()\n",
+    "\n",
+    "        # The initial model, spectral axis (in km/s) and spectrum are passed for the fit\n",
+    "        g_fit = fit_g(g_init, spectrum_axis, spec)\n",
+    "\n",
+    "        results[:, y, x] = g_fit.parameters\n",
+    "\n",
+    "    return results\n"
+   ]
+  },
+  {
+   "source": [
+    "Before fitting the whole cube, we will test this function on a single spectrum. The center pixel has a strong CO(2-1) detection, so we will use it for testing. Note that this is the same spectrum \n",
+    "that we use in the [general fitting with spectral-cube tutorial](https://github.com/radio-astro-tools/tutorials/blob/master/spectral_fitting/fitting_with_spectralcube.ipynb).\n",
+    "\n",
+    "Both here and for fitting the whole cube, we will use `spatmasked_cube`, where only the spatial extent of the signal mask is used. This allows the full spectrum to be included in the fit where we expect there to be bright emission."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# Pixel location\n",
+    "y, x = 32, 32\n",
+    "\n",
+    "spec = spatmasked_cube[:, y, x]\n",
+    "\n",
+    "# Slice out the guesses at this location\n",
+    "guesses = [masked_peaktemp[y:y+1, x:x+1].value, masked_moment1[y:y+1, x:x+1].value,\n",
+    "           masked_equivwidth[y:y+1, x:x+1].value]\n",
+    "\n",
+    "# Feed in the data. Our \"subcube\" or chunk is a single spectrum in this case\n",
+    "fit_params = spectrum_fit_gaussian(spatmasked_cube._get_filled_data(fill=np.NaN)[:, y, x], cube.spectral_axis, guesses)\n",
+    "\n",
+    "# subcube will be 3D when fitting the full cube\n",
+    "fit_params = fit_params.squeeze()\n",
+    "\n",
+    "# Convert the output results to an astropy.model\n",
+    "g_fit_spec = models.Gaussian1D(amplitude=fit_params[0].squeeze(),\n",
+    "                               mean=fit_params[1].squeeze(),\n",
+    "                               stddev=fit_params[2].squeeze())\n",
+    "\n",
+    "print(g_fit_spec)\n",
+    "\n",
+    "spec.quicklook()\n",
```

Comment:
even though this is convenient, I might suggest just writing it out instead - `quicklook` is really meant for quicklook, not production, like this